src/OpenTelemetry.Contrib.Instrumentation.GrpcCore/ServerTracingInterceptorOptions.cs

```diff
@@ -0,0 +1,36 @@
+﻿// <copyright file="ServerTracingInterceptorOptions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Contrib.Instrumentation.GrpcCore
+{
+    using OpenTelemetry.Context.Propagation;
+
+    /// <summary>
+    /// Options for the ServerTracingInterceptor.
+    /// </summary>
+    public class ServerTracingInterceptorOptions
+    {
+        /// <summary>
+        /// Gets or sets a value indicating whether gets a flag indicating whether or not to record individual message events.
+        /// </summary>
+        public bool RecordMessageEvents { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets the propagator.
+        /// </summary>
+        public TextMapPropagator Propagator { get; set; } = Propagators.DefaultTextMapPropagator;

```

Comment:
unless there is a need to override propagator for this instrumentation alone, we dont need to expose this public api. Instrumentation will simply use the Propagator.DefaultTextMapPropagator. If user want to change it, they can change the default to their desired one, and all instrumentations will/should respect it.

Reply:
I can make the setter internal then. I use it configure the propagator for the tests (without referencing the actual SDK). In actual code we can use the SDK to set the  Propagators.DefaultTextMapPropagator